### PR TITLE
Improve comments for autodoc

### PR DESCRIPTION
**Pull request by PimMeulensteen**

### CODE REVIEW

File: lir/config/substitution.py [hunk 1]
```diff
@@ -243,30 +248,36 @@ def parse_float(spec: ContextAwareDict, output_path: Path) -> 'FloatHyperparamet
...
+    Example configuration:
 
-    The generated options will have the full path of each file as both name and value.
+    .. code-block:: yaml
 
-    An example configuration is as follows:
-    ```
         hyperparameters:
-          - path: data.provider.path
+        - path: data.provider.path
```
Comment (wowtor) [inline lir/config/substitution.py:267]: deze indent lijkt niet te kloppen met die hieronder

File: lir/data/data_strategies.py [hunk 2]
```diff
@@ -32,21 +32,25 @@ def apply(self, instances: FeatureData) -> Iterable[tuple[FeatureData, FeatureDa
...
+    The input data must contain class labels. Each fold is constructed so that
+    instances from both classes are present in every split.
+
+    This strategy may be registered in a YAML registry as follows:
+
+    .. code-block:: yaml
 
-    The input data should have class labels. This split assigns instances of both classes to each "fold" subset.
+        data_strategies:
+        binary_cross_validation: lir.data.data_strategies.BinaryCrossValidation
```
Comment (wowtor) [inline lir/data/data_strategies.py]: deze indent klopt niet (maar op zich mag dit hele voorbeeld ook wel weg)

File: lir/data/data_strategies.py [hunk 3]
```diff
@@ -32,21 +32,25 @@ def apply(self, instances: FeatureData) -> Iterable[tuple[FeatureData, FeatureDa
...
 
-    In the benchmark configuration YAML, this validation can be referenced as follows:
-    ```
-    splits:
-      strategy: binary_cross_validation
-    ```
+    .. code-block:: yaml
+
+        splits:
+        strategy: binary_cross_validation
```
Comment (wowtor) [inline lir/data/data_strategies.py]:
```suggestion
          strategy: binary_cross_validation
```

File: lir/data/data_strategies.py [hunk 4]
```diff
@@ -82,24 +86,28 @@ def apply(self, instances: FeatureData) -> Iterable[tuple[FeatureData, FeatureDa
...
+    The input data must contain ``source_ids``. All instances originating from the
+    same source are assigned to the same fold, ensuring that no source appears in
+    both the training and test sets within a split.
+
+    This strategy may be registered in a YAML registry as follows:
+
+    .. code-block:: yaml
+
+        data_strategies:
+        multiclass_cross_validation: lir.data.data_strategies.MulticlassCrossValidation
```
Comment (wowtor) [inline lir/data/data_strategies.py]:
```suggestion
```

File: lir/data/data_strategies.py [hunk 5]
```diff
@@ -82,24 +86,28 @@ def apply(self, instances: FeatureData) -> Iterable[tuple[FeatureData, FeatureDa
...
+
+    In a benchmark configuration, the split strategy can be referenced as:
+
+    .. code-block:: yaml
+
+        data:
+        [...]
+        splits:
```
Comment (wowtor) [inline lir/data/data_strategies.py]:
```suggestion
          [...]
          splits:
```

**Review by wowtor — APPROVED** [review]
Mooi!

Misschien nog even de indents checken. En soms zit er uitleg bij hoe je iets aan de registry toevoegt, dat mag van mij ook helemaal weg in plaats van fixen.
